circuits/cpp/barretenberg/ts/package.json

```diff
@@ -1,51 +1,61 @@
-    "bb": "dest/node/main.js"
+    "bb.js": "./dest/node/main.js"
   },
   "files": [
     "src/",
     "dest/",
-    "!dest/browser/*.wasm",
     "README.md"
   ],
   "scripts": {
-    "clean": "rm -rf ./dest .tsbuildinfo.browser .tsbuildinfo.node",
-    "build": "yarn clean && yarn build:wasm && yarn build:ts:browser && yarn build:ts:node",
+    "clean": "rm -rf ./dest .tsbuildinfo",
+    "build": "yarn clean && yarn build:wasm && yarn build:ts",
     "build:dev": "tsc -b --watch",
-    "build:wasm": "cd ../cpp && cmake --preset wasm-threads && cmake --build --preset wasm-threads && cmake --preset wasm && cmake --build --preset wasm",
-    "build:ts:browser": "tsc -b tsconfig.browser.json && BUILD_TARGET=browser webpack && chmod +x ./dest/browser/main.js && BUILD_TARGET=browser node replace_imports.cjs",
-    "build:ts:node": "tsc -b tsconfig.node.json && chmod +x ./dest/node/main.js && BUILD_TARGET=node node replace_imports.cjs",
+    "build:wasm": "cd ../cpp && cmake --preset wasm-threads && cmake --build --preset wasm-threads && cmake --preset wasm && cmake --build --preset wasm && ./scripts/strip-wasm.sh",
+    "build:ts": "tsc -b && cp ./src/barretenberg-threads.wasm ./dest/node && webpack && chmod +x ./dest/node/main.js",
     "build:bindings": "cd .. && ./scripts/bindgen.sh",
-    "serve": "webpack serve",
     "formatting": "prettier --check ./src && eslint --max-warnings 0 ./src",
     "formatting:fix": "prettier -w ./src",
-    "test": "yarn build:ts:browser && yarn build:ts:node && yarn test:jest && yarn test:bin",
-    "test:jest": "NODE_NO_WARNINGS=1 node --experimental-vm-modules $(yarn bin jest) --no-cache --passWithNoTests",
-    "test:bin": "cd ./bin-test && ./bin-test.sh",
-    "test-debug": "NODE_NO_WARNINGS=1 node --inspect-brk=0.0.0.0 --experimental-vm-modules $(yarn bin jest) --no-cache --passWithNoTests --runInBand",
+    "test": "NODE_OPTIONS='--loader ts-node/esm' NODE_NO_WARNINGS=1 node --experimental-vm-modules $(yarn bin jest) --no-cache --passWithNoTests",
+    "test-debug": "NODE_OPTIONS='--loader ts-node/esm' NODE_NO_WARNINGS=1 node --inspect-brk=0.0.0.0 --experimental-vm-modules $(yarn bin jest) --no-cache --passWithNoTests --runInBand",
+    "simple_test": "NODE_OPTIONS='--loader ts-node/esm' NODE_NO_WARNINGS=1 node ./src/examples/simple.rawtest.ts",
     "prepack": "yarn build",
     "deploy": "npm publish --access public"
   },
   "jest": {
-    "testRegex": "./dest/node/.*\\.test\\.js$",
-    "rootDir": "./dest"
+    "preset": "ts-jest/presets/default-esm",
+    "transform": {
+      "./src/.*\\.ts": [
+        "ts-jest",
+        {
+          "useESM": true
+        }
+      ]
+    },
+    "moduleNameMapper": {
+      "^(\\.{1,2}/.*)\\.js$": "$1"
+    },
+    "testRegex": "./src/.*\\.test\\.ts$",
+    "rootDir": "./src"
   },
   "dependencies": {
     "comlink": "^4.4.1",
     "commander": "^10.0.1",
     "debug": "^4.3.4",
-    "idb-keyval": "^6.2.1"
+    "ts-node": "^10.9.1",
+    "tslib": "^2.4.0"
```

Comment:
need to run it locally to see why, but interesting we cannot have these as dev-dependencies

Reply:
tslib needed as per "importHelpers" in tsconfig.json set to true.
Can't justify ts-node off top of my head so maybe error. Will move to dev.